src/lib/AllocationRequestCbor.sol

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.25;
+
+import {CBOR} from "solidity-cborutils/contracts/CBOR.sol";
+import {CBORDecoder} from "filecoin-solidity/utils/CborDecode.sol";
+import {FilecoinCBOR} from "filecoin-solidity/cbor/FilecoinCbor.sol";
+import {CommonTypes} from "filecoin-solidity/types/CommonTypes.sol";
+
+struct AllocationRequestData {
+    uint64 provider;
+    CommonTypes.Cid dataCID;
+    uint64 size;
+    int64 termMin;
+    int64 termMax;
+    int64 expiration;
+}
+
+library AllocationRequestCbor {
+    function encodeRequestData(
+        AllocationRequestData[] memory requests
+    ) internal pure returns (bytes memory) {
+        // Create a new CBOR buffer with an initial capacity.
+        CBOR.CBORBuffer memory buf = CBOR.create(requests.length * 64);
```

Comment:
I don't think that's right. It doesn't include the overhead of encoding array type tags and arrays length. It will be at least additional 3 bytes, more if array length requires more bytes to encode.
See comments here, they include some tips: https://github.com/fidlabs/round-robin-autoallocator/pull/1#discussion_r1991729478